crates/viewer/re_context_menu/src/collapse_expand.rs

```diff
@@ -0,0 +1,88 @@
+//! Logic that implements the collapse/expand functionality.
+//!
+//! This is separated from the corresponding context menu action, so it may be reused directly, in
+//! particular in tests.
+
+use re_entity_db::{EntityDb, InstancePath};
+use re_viewer_context::{CollapseScope, ContainerId, Contents, ViewId, ViewerContext};
+use re_viewport_blueprint::ViewportBlueprint;
+
+pub fn collapse_expand_container(
+    ctx: &ViewerContext<'_>,
+    blueprint: &ViewportBlueprint,
+    container_id: &ContainerId,
+    scope: CollapseScope,
+    expand: bool,
+) {
+    blueprint.visit_contents_in_container(container_id, &mut |contents, _| {
+        match contents {
+            Contents::Container(container_id) => scope
+                .container(*container_id)
+                .set_open(ctx.egui_ctx, expand),
+
+            // IMPORTANT: don't call process_view() here, or the scope information would be lost
```

Comment:
now that this is moved out it's not clear immediately to which `process_view` this refers to

Reply:
Dead ~code~ comment from an intermediate refactoring with this pr. Deleted.